docs/dev-client-gameplay.md

```diff
@@ -253,475 +454,702 @@ const canBeSpawnPoint = Waypoint.flags[eid] & WaypointFlags.canBeSpawnPoint;
-      deletable
-      grabbable={{ cursor: true, hand: true }}
-      destroyAtExtremeDistance
-      floatyObject={{
-        flags: FLOATY_OBJECT_FLAGS.MODIFY_GRAVITY_ON_RELEASE,
-        releaseGravity: 0
-      }}
-      networkedFloatyObject={{
-        flags: FLOATY_OBJECT_FLAGS.MODIFY_GRAVITY_ON_RELEASE
-      }}
-      rigidbody={{ collisionGroup: COLLISION_LAYERS.INTERACTABLES, collisionMask: COLLISION_LAYERS.HANDS }}
-      physicsShape={{ halfExtents: [0.22, 0.14, 0.1] }} /* TODO Physics shapes*/
-      scale={[1, 1, 1]}
-    />
-  );
+export type FooParams = {
+  x?: number;
+  y?: number;
+  z?: number;
+};
+
+const DEFAULTS: Required<FooParams> = {
+  x: 0.0,
+  y: 0.0,
+  z: 0.0
+};
+
+export function inflateFoo(
+  world: HubsWorld,
+  eid: number,
+  params: FooParams
+) {
+  params = Object.assign({}, params, DEFAULTS) as Required<FooParams>;
+  addComponent(world, Foo, eid);
+  Foo.x[eid] = params.x;
+  Foo.y[eid] = params.y;
+  Foo.z[eid] = params.z;
+  addObject3DComponent(world, eid, new FooObject3D());
 }
 ```
 
-Although `EntityDef` s are written with `jsx` syntax, this is not `React`. The `jsx` syntax allows us to describe our desired scene graph, entities, and components. The definition is static, and there should be no expectation of &ldquo;re-rendering&rdquo; (as in `React` or `three-fiber`). Semantically, an `EntityDef` is equivalent to a model file with component data. `EntityDef` s are meant to be easy to edit by hand and to version control.
+Inflator function type must be `(world: HubsWorld, eid: number, params?: object) => void`
+for integration with [our `renderAsEntity()` explained later](#renderasentity).
 
-For `network instantiated` entities, `template` functions are grouped together with `permission` information to form a named `Prefab`. More information about `network instantiated` entities can be found in the networking documentation.
+An inflator may create a `Three.js Object3D` object and add it to an entity
+with our `addObject3DComponent()`. Refer to [the later section](#single-object3d-per-an-entity)
+for adding a `Object3D` to an entity.
 
+An inflator may set up multiple components.
 
-<a id="org156c909"></a>
 
-## Creating model files
+### Inflators are synchronous
 
-Equivalently, hubs components can be added to nodes in a GLTF file with the `MOZ_hubs_components` extension.
+Inflator functions must be synchronous functions for simplicity.
 
-The `hubs-blender-exporter` is a Blender add-on that helps artists do this.
+TODO: More describe the benefit from this limitation
 
-Spoke also includes component data in the gltf files that it exports and uploads.
 
+### Default inflators
 
-<a id="org2319bae"></a>
+TODO: Write this section
 
-## Entity creation is synchronous
 
-It is important to realize that `renderAsEntity` is a synchronous function. That is, it will immediately return a valid `EntityID` with a corresponding `Object3D` added to the scene graph.
+## Creating entities

```

Comment:
Remind us what an entity is? “An entity is an identifier pointing to a group of components. It doesn’t contain any direct behavior or data.”

Reply:
Updated, thanks.